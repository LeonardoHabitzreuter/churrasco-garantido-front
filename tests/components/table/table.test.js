import React from 'react'
import { shallow } from 'enzyme'
import Table from '../../../src/components/table'

describe('table component', () => {
  const wrapper = shallow(<Table columns={{ name: "User's name:", age: "User's age:" }} lines={[{ name: 'Jhon', age: 25, sex: 'male' }, { name: 'Hailie' }]} />)

  describe('snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('thead', () => {
    test('should render just one tr', () => {
      expect(wrapper.find('thead').find('tr')).toHaveLength(1)
    })

    test('should render the same amount of th as the columns', () => {
      expect(wrapper.find('tr').find('th')).toHaveLength(2)
    })

    test('should render the ths with the columns values as text', () => {
      expect(wrapper.find('tr').find('th').map(th => th.text())).toEqual(["User's name:", "User's age:"])
    })
  })

  describe('tbody', () => {
    test('should render the same amount of tr as the lines', () => {
      expect(wrapper.find('tbody').find('tr')).toHaveLength(2)
    })

    test('should not render the props of the lines which is not specified in the columns', () => {
      expect(wrapper.find('td').map(td => td.text())).not.toContain('male')
    })

    test('should render the td with empty text when the line does not have a value for the column', () => {
      expect(wrapper.find('td').map(td => td.text())).toContain('')
    })
  })
})
