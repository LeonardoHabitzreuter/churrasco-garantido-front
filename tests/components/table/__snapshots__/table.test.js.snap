// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Table
  bordered={true}
  bsClass="table"
  condensed={true}
  hover={true}
  responsive={true}
  striped={true}
>
  <thead>
    <tr>
      <th
        key="name"
        style={
          Object {
            "textAlign": "center",
            "verticalAlign": "middle",
          }
        }
      >
        User's name:
      </th>
      <th
        key="age"
        style={
          Object {
            "textAlign": "center",
            "verticalAlign": "middle",
          }
        }
      >
        User's age:
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="0"
    >
      <td
        key="0"
        style={
          Object {
            "textAlign": "center",
            "verticalAlign": "middle",
          }
        }
      >
        Jhon
      </td>
      <td
        key="1"
        style={
          Object {
            "textAlign": "center",
            "verticalAlign": "middle",
          }
        }
      >
        25
      </td>
    </tr>
    <tr
      key="1"
    >
      <td
        key="0"
        style={
          Object {
            "textAlign": "center",
            "verticalAlign": "middle",
          }
        }
      >
        Hailie
      </td>
      <td
        key="1"
        style={
          Object {
            "textAlign": "center",
            "verticalAlign": "middle",
          }
        }
      />
    </tr>
  </tbody>
</Table>
`;
